kernel/common/kernel.rb

```diff
@@ -504,7 +504,7 @@ def load(name, wrap=false)
   module_function :load
 
   def loop
-    return to_enum(:loop) unless block_given?
+    return to_enum(:loop) { Float::INFINITY } unless block_given?
```

Comment:
The to_enum(:loop).size will always return nil unless you give him the size in the block. It's the way [to_enum works](http://ruby-doc.org/core-2.2.0/Object.html#method-i-to_enum). It's up to the developer using the to_enum to provide a size calculation, whenever possible.
You can even do something like this:
```
irb(main):001:0> to_enum(:loop) {"XXL"}.size
=> "XXL"
```
hehehe